### PR TITLE
perf(vue3): improve index loading times by only blocking navigation at login

### PR DESCRIPTION
**Pull request by ferferga**
Due to reactivity bugs (which need to be solved for the index to properly load, as the content might change after the Carousel component is mounted), the remaining Carousel components are migrated to Composition API.

### CODE REVIEW

File: frontend/src/components/Layout/Carousel/CarouselProgressBar.vue [hunk 1]
```diff
@@ -4,148 +4,73 @@
...
-      });
+const props = withDefaults(
+  defineProps<{
+    pages: number;
+    currentIndex: number;
+    duration: number;
+    paused: boolean;
+    hoverable?: boolean;
+  }>(),
+  { paused: false, hoverable: false }
```
[the remark this answers is not in the document]
Comment (ferferga) [reply frontend/src/components/Layout/Carousel/CarouselProgressBar.vue:38]: @ThibaultNocchi Good catch!

File: frontend/src/components/Layout/Carousel/CarouselProgressBar.vue [hunk 2]
```diff
@@ -4,148 +4,73 @@
...
+
+  if (index === currentIndex) {
+    classes += 'active';
+
+    if (paused) {
+      classes += ' paused';
     }
+  } else if (index < currentIndex) {
+    classes += 'passed';
   }
```
[the remark this answers is not in the document]
Comment (ferferga) [reply frontend/src/components/Layout/Carousel/CarouselProgressBar.vue:70]: @ThibaultNocchi How? Isn't more unefficient to later convert that array to an string, and remove the commas?

Seems a bit of overengineering to me

[the remark this answers is not in the document]
Comment (ThibaultNocchi) [reply frontend/src/components/Layout/Carousel/CarouselProgressBar.vue:70]: Indeed, join may be a little bit more unefficient than string concats (though join becomes more efficient with the more data you have), at that level it'll be mere microseconds. 

Secondly, that way you don't have to add spaces in your strings depending on its position. Also, you don't have to remove quotes, just push the strings you need in an array, and return arr.join(' ')

File: frontend/src/components/Layout/Carousel/CarouselProgressBar.vue [hunk 3]
```diff
@@ -4,147 +4,63 @@
...
+const expand = computed(() => props.hoverable && !display.mobile.value);
+const animDuration = computed(() => (props.duration / 1000).toString() + 's');
+const defaultBarClasses = Object.freeze([
+  'progress',
+  'd-flex',
+  'align-center',
+  'justify-center'
+]);
+const barClasses = computed(() => {
+  return Array.from({ length: props.pages }).map((_, i) => {
```
Comment (ThibaultNocchi) [inline frontend/src/components/Layout/Carousel/CarouselProgressBar.vue:45]: You can directly return the Array.from instead

```suggestion
const barClasses = computed(() =>
  Array.from({ length: props.pages }).map((_, i) => {
```

**Review by ThibaultNocchi — COMMENTED** [review]
Just a nitpick

**Review by ThibaultNocchi — APPROVED** [review]

**Pull request merged**